larpandoracontent/LArThreeDReco/LArTrackFragments/ClearTrackFragmentsTool.cc

```diff
@@ -52,52 +52,52 @@ bool ClearTrackFragmentsTool::FindTrackFragments(ThreeDTrackFragmentsAlgorithm *
         IteratorList iteratorList;
         this->SelectClearElements(elementList, iteratorList);
 
-        bool particlesMade(false);
-        ClusterList deletedClusters, modifiedClusters, clustersToRemoveFromTensor, clustersToAddToTensor;
-        ClusterSet badClusters;
-
-        for (IteratorList::const_iterator iIter = iteratorList.begin(), iIterEnd = iteratorList.end(); iIter != iIterEnd; ++iIter)
```

Comment:
I think I understand and agree with the simplification here. In the broader picture, the tensor will be updated after the tool makes a change, so this iteration inside the tool main function is not lost by the replacement with iteratorList.front() as the tool would run again and evaluate if more elements are connected by SelectClearElements. Also, the iteratorList is sorted by overlapResult (from GetConnectedElements in LArOverlapTensor and its operator < definition), which makes sense to start with the largest overlap result in the list of elements. And it solves potential problems so I like the simplification.